streaming/src/test/scala/org/apache/spark/streaming/BasicOperationsSuite.scala

```diff
@@ -210,6 +210,18 @@ class BasicOperationsSuite extends TestSuiteBase {
       input.map(_.map(_.toString))
     )
   }
+  
+  test("transform with NULL") {
+    val input = Seq(1 to 4, Seq(), 5 to 8, 9 to 12)
+    testOperation(
+      input,
+      (r: DStream[Int]) => r.transform(rdd => {if (rdd != null && !rdd.isEmpty()) rdd.map(_.toString) else null}),   // RDD.map in transform
+      input.filter(!_.isEmpty).map(_.map(_.toString)),
+      4, 
+      false
+    )
+  }
```

Comment:
nit: The formatting here is pretty weird. Maybe take the transformation method out and separate it out as a `def`, so this method call can be in a single line:

```
def transformMethod(r: DStream[Int]): Option[RDD] = {
  r.transform { rdd => 
    if (rdd != null && !rdd.isEmpty()) rdd.map(_.toString) else null
  }
}

testOperation(input, transformMethod, input.filter(!_.isEmpty).map(_.map(_.toString)), 4, false)
```